lib/pubnub/client.rb

```diff
@@ -320,27 +319,23 @@ def create_state_pools(event)
     end
 
     def setup_httpclient(event_type)
-      if ENV['HTTP_PROXY']
-        hc = HTTPClient.new(ENV['HTTP_PROXY'])
-      else
-        hc = HTTPClient.new
-      end
+      hc = if ENV['HTTP_PROXY']
+             HTTPClient.new(ENV['HTTP_PROXY'])
+           else
+             HTTPClient.new
+           end
 
       case event_type
       when :subscribe_event
-        hc.connect_timeout = @env[:s_open_timeout]
-        hc.send_timeout = @env[:s_send_timeout]
-        hc.receive_timeout = @env[:s_read_timeout]
+        hc.receive_timeout = 310
```

Comment:
Is there a reason these were removed? We'd like to have them restored so we can control the worst case operating time of these calls.

Reply:
@callumj Let me do a research about it and I will get back to you.

Reply:
@callumj I think this will do the things right for you https://github.com/pubnub/ruby/pull/72 ?